### DISCUSSION TITLE
How to speed up loading website to use code-server

### DISCUSSION THREAD

**Discussion by cxw620** [coder/code-server#4892] [Q&A]
Well I'm from China and In China, bandwidth resources are very expensive, causing that VPS's public network bandwidth for individual developers is relatively small. What's more, due to the special network environment in China, open-vsx.org is also slow to access. 
I wonder if code-server supports using CDN to cache static resources to speed up loading website accessing code-server. What's more, if code-server support using a mirror site for open-vsx.org I would be really happy.

(There may be some  grammatical errors for I'm still trying my best learning English😊)

**Comment by jsjoeio** [comment]
> I wonder if code-server supports using CDN to cache static resources to speed up loading website accessing code-server. What's more, if code-server support using a mirror site for open-vsx.org I would be really happy.

Unfortunately, that's not something we have bandwidth for :( Have you tried looking around to see how others have solved this?

**Reply by cxw620** [reply]
> > I wonder if code-server supports using CDN to cache static resources to speed up loading website accessing code-server. What's more, if code-server support using a mirror site for open-vsx.org I would be really happy.
> 
> Unfortunately, that's not something we have bandwidth for :( Have you tried looking around to see how others have solved this?

ECDN may be a possible solution though it still costs a large amount of money :-(

I notice that code-server seems only support **gzip** and **http1.1**. I think utilizing `Brotli Compress` and `http2` may help much. What's more, loading `workbench.*.js`, like `workbench.web.api.js` takes much time. I don't know if it's a static file that won't change unless code-sever updates. If it won't change, it will be really helpful to add a appropriate `Cache-Control` response directives.

PS: Unfortunately, NodeJs doesn't support HTTP/3 now. I found it faster to load code-server if I enable QUIC support when configuring ECDN.
PS: Certainly if I use code-server locally I don't need to try hard to get a lower bandwidth use.
PS again: I use proxy to temporarily solve the second problem. If code-serer would allow users changing where to download plugins in the future, I can have open-vsx.org mirrored in China.  as it's an open-source project, too.

**Reply by jsjoeio** [reply]
> PS again: I use proxy to temporarily solve the second problem. If code-serer would allow users changing where to download plugins in the future, I can have open-vsx.org mirrored in China. as it's an open-source project, too.

Hmm...I don't know how difficult that would be to support. Maybe there's already support in VS Code? If so, could be something we expose. We're not considering many feature requests at the moment (only bug fixes) but you can open a Feature Request to see if other people want it.

**Comment by alanhe421** [comment]
so, What resources can we go to COS-CDN?


for js, I don't know if the following js can be loaded by cdn.

1. /out/vs/loader.js
2. /out/vs/webPackagePaths.js
3. /out/vs/workbench/workbench.web.main.nls.js
 and so on.

<img width="878" alt="image" src="https://user-images.githubusercontent.com/9245110/218390111-546310fd-4569-4e77-aa89-8060013099c5.png">
